server/master/src/main/java/org/apache/accumulo/master/Master.java

```diff
@@ -1336,7 +1332,11 @@ public void assignedTablet(KeyExtent extent) {
   }
 
   @Override
-  public Collection<KeyExtent> migrations() {
-    return migrations.keySet();
+  public Set<KeyExtent> migrations() {
+    Set<KeyExtent> migrationsCopy = new HashSet<KeyExtent>();
+    synchronized (migrations) {
+      migrationsCopy.addAll(migrations.keySet());
+    }
+    return migrationsCopy;
```

Comment:
> We could change the interface method name to "migrationsSnapshot()" to be clear, though, and to avoid careless future impl changes which change that semantics.

:+1: I like this general approach. I think migrations are just one place that we follow this pattern and could benefit from more descriptive method names.